src/compile/repeat.ts

```diff
@@ -32,7 +32,7 @@ export function replaceRepeaterInEncoding(encoding: Encoding<Field>, repeater: R
   return replaceRepeater(encoding, repeater);
 }
 
-type EncodingOrFacet<F> = Encoding<F> | Facet<F>;
+type EncodingOrFacet<F> = Encoding<F> & Facet<F>;
```

Comment:
I already know about your tweet.  

That said, intersection type ("and") is not the thing we really want here. 
The right thing semantically would be 

`EncodingWithFacet<F> | Facet<F>`  as we will take mapping from either unit spec (`EncodingWithFacet`) or from facet spec (`Facet`).   

It just happens that all properties of Encoding and Facet are optional and thus `Encoding<F> & Facet<F>;`  makes TS stop complaining although it is not semantically correct.